Mathlib/Topology/MetricSpace/MetricSeparated.lean

```diff
@@ -1,38 +1,74 @@
 /-
-Copyright (c) 2021 Yury Kudryashov. All rights reserved.
+Copyright (c) 2021 Yury Kudryashov, Yaël Dillies. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
-Authors: Yury Kudryashov
+Authors: Yury Kudryashov, Yaël Dillies
 -/
 import Mathlib.Topology.EMetricSpace.Defs
 
 /-!
-# Metric separated pairs of sets
+# Metric separation
 
-In this file we define the predicate `Metric.AreSeparated`. We say that two sets in an (extended)
-metric space are *metric separated* if the (extended) distance between `x ∈ s` and `y ∈ t` is
-bounded from below by a positive constant.
+This file defines a few notions of separations of sets in a metric space.
 
-This notion is useful, e.g., to define metric outer measures.
--/
+## Main declarations
 
+* `Metric.IsSeparated`: A set `s` is `ε`-separated if its elements are pairwise at distance at least
+  `ε` from each other.
+* `Metric.AreSeparated`: Two sets `s` and `t` are separated if the distance between `x ∈ s` and
+  `y ∈ t` is bounded from below by a positive constant.
+-/
 
 open EMetric Set
+open scoped ENNReal
 
 noncomputable section
 
 namespace Metric
+variable {X : Type*} [PseudoEMetricSpace X] {s t : Set X} {ε δ : ℝ≥0∞} {x : X}
+
+/-!
+### Metric-separated sets
+
+In this section we define the predicate `Metric.IsSeparated` for `ε`-separated sets.
+-/
+
+/-- A set `s` is `ε`-separated if its elements are pairwise at distance at least `ε` from each
+other. -/
+def IsSeparated (ε : ℝ≥0∞) (s : Set X) : Prop := s.Pairwise (ε < edist · ·)
+
+protected lemma IsSeparated.empty : IsSeparated ε (∅ : Set X) := pairwise_empty _
+protected lemma IsSeparated.singleton : IsSeparated ε {x} := pairwise_singleton ..
+
+@[simp] lemma IsSeparated.of_subsingleton (hs : s.Subsingleton) : IsSeparated ε s := hs.pairwise _
+
+alias _root_.Set.Subsingleton.isSeparated := IsSeparated.of_subsingleton
+
+nonrec lemma IsSeparated.anti (hεδ : ε ≤ δ) (hs : IsSeparated δ s) : IsSeparated ε s :=
```

Comment:
Do you want to make some `Antitone`? I don't have a strong feeling either way.

Reply:
I think it would be more confusing than helpful

Reply:
fair